backends/mongo.go

```diff
@@ -92,11 +106,21 @@ func NewMongo(authOpts map[string]string, logLevel log.Level) (Mongo, error) {
 	opts.ApplyURI(addr)
 
 	if m.Username != "" && m.Password != "" {
-		opts.Auth = &options.Credential{
-			AuthSource:  m.DBName,
-			Username:    m.Username,
-			Password:    m.Password,
-			PasswordSet: true,
+		if m.AuthSource != "" {
+			log.Infof("Mongo set AuthSource to: %s", m.AuthSource)
+			opts.Auth = &options.Credential{
+				AuthSource:  m.AuthSource,
+				Username:    m.Username,
+				Password:    m.Password,
+				PasswordSet: true,
+			}
+		} else {
+			opts.Auth = &options.Credential{
+				AuthSource:  m.DBName,
+				Username:    m.Username,
+				Password:    m.Password,
+				PasswordSet: true,
+			}
```

Comment:
Similarly, you could just predefine `opts.Auth` and then assign its `AuthSource` conditionally.

Reply:
Modified approach in latest commit.